packages/@aws-cdk/aws-apigateway/lib/integrations/lambda.ts

```diff
@@ -60,14 +61,14 @@ export class LambdaIntegration extends AwsIntegration {
     this.handler.addPermission(`ApiPermission.${desc}`, {
       principal,
       sourceArn: method.methodArn,
-    });
+    }, method.node.scope as cdk.Construct);
```

Comment:
Why not `method`? It's usually a bad practice to add constructs to scopes you don't "own". What happens if the parent scope has two methods? The namespace is not "clean"...

Reply:
All good points. Fixing up.